manifests/puppetserver.pp

```diff
@@ -16,6 +16,33 @@
     metric_ensure => $metrics_ensure,
     hosts         => $hosts,
     metrics_port  => $port,
-  }
+    additional_metrics => $additional_metrics,
 
-}
+  }
+  
+  if versioncmp($facts['pe_server_version'], '2018.1.0') < 0 {
+    $additional_metrics = [
+      { 'name' => 'compiler.find_node',
+        'url'  => "puppetserver:name=puppetlabs.${::hostname}.compiler.find_node" },
+      { 'name' => 'puppetdb.query',
+        'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.query" },
+      { 'name' => 'puppetdb.resource.search',
+        'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.resource.search" },
+      { 'name' => 'puppetdb.facts.encode',
+        'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.facts.encode" },
+      { 'name' => 'puppetdb.command.submit.replace facts',
+        'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.command.submit.replace facts" },
+      { 'name' => 'puppetdb.catalog.munge',
+        'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.catalog.munge" },
+      { 'name' => 'puppetdb.command.submit.replace catalog',
+        'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.command.submit.replace catalog" },
+      { 'name' => 'puppetdb.report.convert_to_wire_format_hash',
+        'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.report.convert_to_wire_format_hash" },
+      { 'name' => 'puppetdb.command.submit.store report',
+        'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.command.submit.store report" },
+      ],
```

Comment:
Since this is the end of a variable assignment, the trailing comma is causing a parse failure. Would be appropriate for resource parameters, but doesn't work with assignment.